accumulator/batchproof.go

```diff
@@ -7,24 +7,37 @@ import (
 	"io"
 )
 
-// BatchProof :
+// BatchProof is the inclusion-proof for multiple leaves.
 type BatchProof struct {
+	// Targets are the ist of leaf locations to delete. These are the bottommost leaves.
+	// With the tree below, the Targets can only consist of one of these: 00, 01, 02, 03
+	//
+	// 06
+	// |-------\
+	// 04      05
+	// |---\   |---\
+	// 00  01  02  03
 	Targets []uint64
-	Proof   []Hash
-	// list of leaf locations to delete, along with a bunch of hashes that give the proof.
-	// the position of the hashes is implied / computable from the leaf positions
-}
 
-/*
-Batchproof serialization is:
-4bytes numTargets
-4bytes numHashes
-[]Targets (8 bytes each)
-[]Hashes (32 bytes each)
-*/
+	// All the nodes in the tree that are needed to hash up to the root of
+	// the tree. If Targets are [00, 01], then Proof would be [00, 01, 04, 05, 06].
```

Comment:
The proof for [00, 01] would actually just be [00, 01, 05] the other values you can compute from those.

Reply:
Ideally you wouldnt even need the targets in there so just [05].

Reply:
Ah ok thanks. Will fix